src/utilitis/logger.ts

```diff
@@ -1,52 +1,105 @@
+ * export HUDDLY_LOG_LEVEL=INFO
+ * node sdkexample.js
+ *
+ * // Disable all log output
+ * export HUDDLY_LOG_LEVEL=NONE
+ * node sdkexample.js
+ */
 export default class Logger {
-  _verbose: boolean;
+  static customLogger: ILogger = undefined;
 
-  constructor(verbose: boolean) {
-    this._verbose = verbose;
+  static setLogger(logger: ILogger): void {
+    Logger.customLogger = logger;
   }
 
-  warn(message: string, component: string = 'Generic Component'): void {
-    if (this.verbose) {
-      console.warn(this.formatLogMsg('WARN', component, message));
-    }
-  }
+  static warn(message: string, component: string = 'Generic Component'): void {
+    if (['DEBUG'].indexOf(process.env.HUDDLY_LOG_LEVEL) > -1) {
+      if (Logger.customLogger) {
+        Logger.customLogger.warn(message);
+        return;
+      }
 
-  info(message: string, component: string = 'Generic Component'): void {
-    if (this.verbose) {
-      console.log(this.formatLogMsg('INFO', component, message));
+      this.redirectLogMsg(this.formatLogMsg('WARN', component, message));
     }
   }
 
-  debug(message: string, component: string = 'Generic Component'): void {
-    if (this.verbose) {
-      console.log(this.formatLogMsg('DEBUG', component, message));
+  static info(message: string, component: string = 'Generic Component'): void {
+    if (['INFO', 'DEBUG'].indexOf(process.env.HUDDLY_LOG_LEVEL) > -1) {
+      if (Logger.customLogger) {
+        Logger.customLogger.info(message);
+        return;
+      }
+
+      this.redirectLogMsg(this.formatLogMsg('INFO', component, message));
     }
   }
 
-  error(message: string, stackTrace: any, component: string = 'Generic Component'): void {
-    if (this.verbose) {
-      console.error(this.formatLogMsg('ERROR', component, message));
-      console.error(stackTrace);
+  static debug(message: string, component: string = 'Generic Component'): void {
+    if (['DEBUG'].indexOf(process.env.HUDDLY_LOG_LEVEL) > -1) {
+      if (Logger.customLogger) {
+        Logger.customLogger.debug(message);
+        return;
+      }
+
+      this.redirectLogMsg(this.formatLogMsg('DEBUG', component, message));
     }
   }
 
-  get verbose(): boolean {
-    return this._verbose;
+  static error(message: string, stackTrace: any, component: string = 'Generic Component'): void {
+    if (['NONE'].indexOf(process.env.HUDDLY_LOG_LEVEL) == -1) {
+      if (Logger.customLogger) {
+        Logger.customLogger.error(message, stackTrace);
+        return;
+      }
+
+      this.redirectLogMsg(this.formatLogMsg('ERROR', component, message));
+      this.redirectLogMsg(stackTrace);
+    }
   }
 
-  set verbose(verbose: boolean) {
-    this._verbose = verbose;
+  static redirectLogMsg(message: string): void {
+    if (process.env.HUDDLY_LOG_CHANNEL === 'FILE') {
+      const logFile: string = process.env.HUDDLY_LOG_FILE || 'huddlysdk.log';
+      fs.appendFileSync(logFile, `${message}\n`);
+    } else {
+      console.log(message);
```

Comment:
Is this necessary anymore. Could just have console as the default customLogger?

Reply:
@jontore you mean logging into a file if it's necessary?

Reply:
We can easily remove it later on if we want to. Default is console with this revision anyways. I dont think it hurts to keep the alternative of directing the logs to a file if desired.